Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/validation/core/server/DccValidationServer.kt

```diff
@@ -24,17 +30,27 @@ class DccValidationServer @Inject constructor(
     private val rulesApi: Lazy<DccValidationRuleApi>,
     @CertificateValidation private val cache: Cache,
     private val signatureValidation: SignatureValidation,
+    private val dispatcherProvider: DispatcherProvider,
 ) {
 
     private val dccValidationRuleApi: DccValidationRuleApi
         get() = rulesApi.get()
 
-    suspend fun ruleSet(ruleTypeDcc: DccValidationRule.Type): Set<DccValidationRule> {
-        return when (ruleTypeDcc) {
-            DccValidationRule.Type.ACCEPTANCE -> dccValidationRuleApi.acceptanceRules()
-            DccValidationRule.Type.INVALIDATION -> dccValidationRuleApi.invalidationRules()
+    suspend fun ruleSetJson(ruleTypeDcc: DccValidationRule.Type): String =
+        withContext(dispatcherProvider.Default) {
```

Comment:
We shouldn't explicitly switch to the `Default`, either no switch (and let the caller decide) or switch to `IO` (probably best). Due to the amount of threads in `Default` being capped, a http request timing out could block a thread for 10-20 seconds potentially starving other coroutines.